mllib/src/test/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifierSuite.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.classification
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.mllib.linalg.Vectors
+import org.apache.spark.mllib.regression.LabeledPoint
+import org.apache.spark.mllib.util.MLlibTestSparkContext
+import org.apache.spark.sql.Row
+
+class MultilayerPerceptronClassifierSuite extends SparkFunSuite with MLlibTestSparkContext {
+
+  test("XOR function learning as 2-class classification problem") {
+    val inputs = Array[Array[Double]](
+      Array[Double](0, 0),
+      Array[Double](0, 1),
+      Array[Double](1, 0),
+      Array[Double](1, 1)
+    )
+    val outputs = Array[Double](0, 1, 1, 0)
+    val data = inputs.zip(outputs).map{ case(input, output) =>
```

Comment:
You can construct the DataFrame directly:

~~~scala
val df = sqlContext.createDataFrame(Seq(
  (Vectors.dense(0.0, 0.0), 0.0),
  (Vectors.dense(0.0, 1.0), 1.0),
  ...)
).toDF("features", "label")
~~~